src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -1916,6 +1917,27 @@ public void afterSave() {
         }
     }
 
+    /**
+     * This extension intends to perform a GET request without any credentials on the provided repository URL
+     * to return the size of repository.
+     */
```

Comment:
Why do this?

Reply:
Git Plugin is adding a new class called the `GitToolChooser` which requires the size of a remote repository to recommend an optimal git implementation ("git" or "jgit").

This size can be measured in two ways: 
- By cache created by a multi branch project
- From REST APIs exposed by the git providers themselves

Since the git plugin probably doesn't involve calling any REST API in the code base, we would like to delegate the task to the plugins which do. 

In terms of the GitHub Branch Source Plugin, implementing this extension serves no purpose for the plugin, it only acts as a helper method for git plugin to improve internal performance.